### ISSUE TITLE
Line Object value x1/y1/x2/y2 is not updated when moving object

### ISSUE THREAD

**Issue by ncou** [fabricjs/fabric.js#4273]
Hi,

The x1/y1/x2/y2 value doesn't seems to be updated when i move my object.
Am i missing something ?

http://jsfiddle.net/tmchz9h0/14/

PS : I try to make a circle follow each part of the line, so i need the updated values to move the circles.

Thanks for you help.

**Comment by asturur** [comment]
The x1/y1/x2/y2 are used for constructor and then are there to do... nothing.
I have no idea if we should fix this.

We cannot implement it on moving since it would clutter the move event without reason, but we can totally write a short method `get current points` that works for line, polygons, polylines an paths, and will give you them back updated.

the x1,y1 is aleays -width/2 -height/2 in the line, while x1,y2 is the opposite. Then you need to transform this 2 points with the calcTransformMatrix of the object.

**Comment by gaooyao** [comment]
@asturur Function 'getCurrentPoint' is very useful for me.I need caculate the polygon's final point on the canvas and send them to machine tool.I hope this function.Thank you.

**Comment by samael205** [comment]
Hi, Can be this another solution?


See [PositoinBTN Example](http://jsfiddle.net/tmchz9h0/27/)

**Comment by asturur** [comment]
x1,y1 are never updated @r0b1n1sl4m.

You can calculate their current position tho.

**Comment by surajwaghmare** [comment]
Hi,

You guys can get those values by following code:

`
       lineObj.toObject = function () {
            return {
                x1: lineObj.left,
                y1: lineObj.top,
                x2: lineObj.left + (lineObj.x2 - lineObj.x1),
                y2: lineObj.top + (lineObj.y2 - lineObj.y1)
            };
        };
`

**Comment by advename** [comment]
> Hi,
> 
> You guys can get those values by following code:
> 
> `lineObj.toObject = function () { return { x1: lineObj.left, y1: lineObj.top, x2: lineObj.left + (lineObj.x2 - lineObj.x1), y2: lineObj.top + (lineObj.y2 - lineObj.y1) }; };`

This is not correct as it assumes that the line starts in top left and ends at bottom right.

**Comment by technorior** [comment]
For moving line with originX: 'center' and originY: 'center'  

```
lineObj.toObject = () => { return {
            x1: lineObj.left - (lineObj.x2 - lineObj.x1)/2,
            y1: lineObj.top - (lineObj.y2 - lineObj.y1)/2,
            x2: lineObj.left + (lineObj.x2 - lineObj.x1)/2,
            y2: lineObj.top + (lineObj.y2 - lineObj.y1)/2 }; };
```
